main/inc/lib/myspace.lib.php

```diff
@@ -1156,137 +1156,253 @@ public static function displayResumeLP(
+                if (!isset($users[$authorData])) {
+                    $users[$authorData] = api_get_user_info($authorData);
                 }
-                $studentUsers = [];
-                for ($i = 0; $i < count($registeredUsers); $i++) {
-                    $studentUsers[] = api_get_user_info($registeredUsers[$i]);
-                }
-                $teacherList = [];
-                for ($i = 0; $i < count($teachersId); $i++) {
-                    $teacherId = $teachersId[$i];
-                    $teacher = api_get_user_info($teacherId);
-                    $data[$teacher['complete_name']][$lpInfo['name']] = [
-                        'students' => count($studentUsers),
-                        'studentList' => $studentUsers,
-                        'lpInfo' => $lpInfo,
-                    ];
-                    $teacherList[] = $teacher;
+                $teachers[$authorData][$lpId] = $users[$authorData];
+                $learningPaths[$lpId]['teachers'][$authorData] = $users[$authorData];
+            } else {
+                $items = explode(',', $authorData);
+                for ($j = 0; $j < count($items); $j++) {
+                    $authorData = $items[$j];
+                    if (!isset($users[$authorData])) {
+                        $users[$authorData] = api_get_user_info($authorData);
+                    }
+                    $teachers[$authorData][$lpId] = $users[$authorData];
+                    $learningPaths[$lpId]['teachers'][$authorData] = $users[$authorData];
                 }
             }
         }
-        if ($csv == false) {
-            $table = "<div class='table-responsive'>".
+        $lpItems = array_unique($lpItems);
+        $whereInLp = implode(',', $lpItems);
+        if (count($lpItems) != 0) {
+            $registeredUsers = self::getCompanyLearnpathSubscription(
+                $startDate,
+                $endDate,
+                $whereInLp);
+            foreach ($registeredUsers as $student) {
+                $totalStudents = count($student);
+                for ($i = 0; $i < $totalStudents; $i++) {
+                    $user = $student[$i];
+                    $lpId = $user['lp_item'];
+                    $studentId = $user['id'];
+                    if (!isset($users[$studentId])) {
+                        $users[$studentId] = api_get_user_info($studentId);
+                        $users[$studentId]['company'] = $user['company'];
+                    }
+                    $learningPaths[$lpId]['courseStudent'][$studentId] = $users[$studentId];
+                }
+            }
+
+            $registeredUsersBySession = self::getSessionAddUserCourseFromTrackDefault(
+                $startDate,
+                $endDate,
+                $whereInLp
+            );
+            foreach ($registeredUsersBySession as $lpId => $student) {
+                $totalStudents = count($student);
+                for ($i = 0; $i < $totalStudents; $i++) {
```

Comment:
Usar foreach

Reply:
comentario añadido "use 'for' to performance"